state/factory/factory.go

```diff
@@ -122,6 +134,32 @@ func InMemTrieOption() Option {
 	}
 }
 
+func newTwoLayerTrie(rootKey string, dao db.KVStore, create bool) (*TwoLayerTrie, error) {
+	_, err := dao.Get(ArchiveTrieNamespace, []byte(rootKey))
+	switch errors.Cause(err) {
+	case db.ErrNotExist:
+		if !create {
+			return nil, err
+		}
+	case nil:
+		break
+	default:
+		return nil, err
+	}
+	dbForTrie, err := db.NewKVStoreForTrie(ArchiveTrieNamespace, dao)
```

Comment:
so if the kvstore exist already, this `NewKVStoreForTrie` doesn't have any side effect?